src/Combobox/index.js

```diff
@@ -0,0 +1,235 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import cc from "classcat";
+import iconSelection from "src/icons/selection.json";
+import { useCombobox } from "downshift";
+import ComboboxItem from "./ComboboxItem";
+import ComboboxHeading from "./ComboboxHeading";
+import TextInput from "../TextInput";
+
+const noop = () => {};
+
+export const VALID_ICON_NAMES = iconSelection.icons.map(
+  (icon) => icon.properties.name
+);
+
+/**
+ * @param {Object} component a Combobox.Item or Combobox.Heading component
+ * @returns {Boolean} true if the item is a selectable Combobox.Item
+ */
+export const isSelectable = (component) => {
+  let result = false;
+  if (component) {
+    result = "value" in component.props;
+  }
+  return result;
+};
+
+/**
+ * Autocomplete input component following the accessible
+ * [ARIA combobox pattern](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role).
+ *
+ * Autocomplete options are defined by passing the `Combobox.Item` subcomponent
+ * as children. To add heading dividers, use the `Combobox.Heading` subcomponent.
+ *
+ * By default options will be filtered down as the user types in the input. This
+ * behavior can be disabled via the `disableFiltering` prop.
+ */
+const Combobox = ({
+  label,
+  onChange = noop,
+  onInputChange = noop,
+  children,
+  disableFiltering = false,
+  errorText,
+  icon,
+  testId,
+}) => {
+  // It is possible that a consumer may have nothing to pass to `children`.
+  // For example, if an API response hasn't completed to load in the autocomplete
+  // options. In that case, Cobmobox should render a normal TextInput.
+  if (React.Children.toArray(children) < 1) {
+    return (
+      <TextInput
+        error={errorText}
+        label={label}
+        value={inputValue}
+        startIcon={icon}
+        onChange={onInputChange}
+      />
+    );
+  }
```

Comment:
Hmm, how would you feel about going through the full downshift stuff even for empty arrays? I don't think inputValue is defined just yet, and we'll probably want to still be firing onInputValueChanged() in case this is something that's doing some debounced server call to load in options based on what the user has started to type? :thinking:

Reply:
Good point - I think I need to pass something different to `value` here either way

Reply:
Fixed - and added a story to clarify the behavior

<img width="1052" alt="Screen Shot 2022-06-13 at 6 48 49 PM" src="https://user-images.githubusercontent.com/231252/173460238-48a90f41-f041-4c81-b001-66aaa8874144.png">